### PR TITLE
Make cache version configurable

### PR DESCRIPTION
**Pull request by lstrojny**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

This PR allows setting a custom cache version. Backwards compatibility is fully preserved. This is for symfony/symfony#25958

### CODE REVIEW

**Review by nicolas-grekas — COMMENTED** [review]
I don't understand this PR: the build_id is sourced by the build_time parameter, which you are making reproducible in another PR. This means the version will also be reproducible. No need to change anything here AFAIK. The less config options the better.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd [hunk 1]
```diff
@@ -247,6 +247,8 @@
             <xsd:element name="default-memcached-provider" type="xsd:string" minOccurs="0" maxOccurs="1" />
             <xsd:element name="pool" type="cache_pool" minOccurs="0" maxOccurs="unbounded" />
         </xsd:sequence>
+        <xsd:attribute name="version" type="xsd:string"/>
+        <xsd:attribute name="prefix-seed" type="xsd:string"/>
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: adding this missing attribute in the XSD is a bug fix though. Can you submit a PR to the 3.4 branch with this change only ?

Comment (lstrojny) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: Changed

**Comment by lstrojny** [comment]
@nicolas-grekas the idea here is that one is able to independently change the cache version based on _whatever_. That’s why I've added the config version. But I am wondering if the prefix seed wouldn't be enough.

**Comment by nicolas-grekas** [comment]
> the prefix seed wouldn't be enough

I think it is, that's its purpose :)

**Comment by lstrojny** [comment]
Alright, let’s close it for now

**Pull request closed without merging**